### ISSUE TITLE
Possible file descriptor leak

### ISSUE THREAD

**Issue by ad-m-ss** [awslabs/iam-roles-anywhere-session#1]
Hi,

I notice that your code use `open` without context manager: https://github.com/awslabs/iam-roles-anywhere-session/blob/25f84e1ffc3f1b7a5639ba8fd19f8515f3041402/iam_rolesanywhere_session/iam_rolesanywhere_session.py#L223-L225

Is there any reason for that? It creates risk of file description leak when parsing of file crashing. I suggest rewrite it to something like:

```python
with open(private_key, 'rb') as key_pem_file:
   return crypto.load_privatekey(crypto.FILETYPE_PEM, key_pem_file.read(), passphrase=self.private_key_passphrase)
```

CC: @tbuatois based on git blame

**Comment by tbuatois** [comment]
Hello @ad-m-ss , 

I want to thank you for your time and your constructive feedback, indeed your implementation follows the best practices. 
I will make sure to add this in the next release.

**Comment by tbuatois** [comment]
fIxed in v2.0.0
